tests/integration/test_ingress_traefik.py

```
@@ -44,10 +45,12 @@ async def test_setup(ops_test, parca_charm, parca_resources):
 
 
 def _get_ingress_ip(model_name):
-    result = getoutput(
-        f"sudo microk8s.kubectl -n {model_name} get svc/{TRAEFIK}-lb -o=jsonpath='{{.status.loadBalancer.ingress[0].ip}}'"
-    )
-    return result.strip("'")
+    cmd = f"microk8s.kubectl -n {model_name} get svc/{TRAEFIK}-lb -o=jsonpath='{{.status.loadBalancer.ingress[0].ip}}'"
+    try:
+        proc = subprocess.run(shlex.split(cmd), text=True, capture_output=True)
+    except:
+        proc = subprocess.run(shlex.split("sudo " + cmd), text=True, capture_output=True)
```

Should this be another try-except? or are we okay with it breaking in case there's something else wrong with kubectl since we're in tests anyway?

---

yes I do want it to fail loudly in this case. I wonder if we can be more specific about the error we're catching in the first place